### PR TITLE
removed GrpcUtil.IS_RESTRICTED_APPENGINE and dependent code

### LINKED ISSUE #6045: Delete GrpcUtil.IS_RESTRICTED_APPENGINE
IS_RESTRICTED_APPENGINE was for GAE Java 7. That doesn't exist any more, so these code paths are dead code. We should delete IS_RESTRICTED_APPENGINE and assume it is always `false`.

### PR DESCRIPTION
**Pull request by dharmeshjogadia**
fixed #6045

### CODE REVIEW

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* One or more commits are not associated with a GitHub user. The email used in the Git commit must match a verified email address in a GitHub user profile. For more information, please see https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/

_Regards,
CLA GitHub bot_

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @dharmeshjogadia isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
CLA GitHub bot_

**Comment by dharmeshjogadia** [comment]
I signed it

File: core/src/main/java/io/grpc/internal/GrpcUtil.java [hunk 1]
```diff
@@ -569,16 +551,11 @@ public void close(ScheduledExecutorService instance) {
...
-      @SuppressWarnings("BetaApi")
-      ThreadFactory factory = MoreExecutors.platformThreadFactory();
-      return factory;
-    } else {
-      return new ThreadFactoryBuilder()
-          .setDaemon(daemon)
-          .setNameFormat(nameFormat)
-          .build();
-    }
+
```
Comment (creamsoup) [inline core/src/main/java/io/grpc/internal/GrpcUtil.java]: nit: don't need extra space here.

**Comment by creamsoup** [comment]
merged, thanks @dharmeshjogadia!

**Pull request merged**